### DISCUSSION TITLE
Placing coloured text over B&W image

### DISCUSSION THREAD

**Discussion by apandzo** [ImageMagick/ImageMagick#6141] [Digital Image Processing] [answered]
Hi guys,
Sorry, I am new to ImageMagick. I am trying to place coloured text as a watermark on a B&W image, but my text comes as black on some of the images. I would appreciate it if you could please advise what can I do to have it as coloured text. I am using Nuget and C#. This is my code below:

                                                `  // Text label watermark settings
                                                using var watermark = new MagickImage();
                                                var labelSettings = new MagickReadSettings();
                                                labelSettings.BackgroundColor = new MagickColor(MagickColors.Transparent);
                                                labelSettings.Font = textFont; //This is my variable
                                                labelSettings.FontPointsize = fontSize; //This is my variable

                                                if (textFontType == "B")
                                                    labelSettings.FontStyle = FontStyleType.Bold;

                                                if (textFontType == "I")
                                                    labelSettings.FontStyle = FontStyleType.Italic;

                                                labelSettings.FillColor = new MagickColor(textColor);
                                                
                                                var gravity = Gravity.Undefined;
                                                if (!string.IsNullOrWhiteSpace(watermarkPlacement))
                                                {
                                                    gravity = watermarkPlacement switch
                                                    {
                                                        "TL" => Gravity.Northwest,
                                                        "T" => Gravity.North,
                                                        "TR" => Gravity.Northeast,
                                                        "L" => Gravity.West,
                                                        "R" => Gravity.East,
                                                        "BL" => Gravity.Southwest,
                                                        "B" => Gravity.South,
                                                        "BR" => Gravity.Southeast,
                                                        "C" => Gravity.Center,
                                                        _ => Gravity.Center
                                                    };
                                                }

                                                // Create the label image.
                                                watermark.Read("label:" + text, labelSettings); //text is my variable
                                                watermark.Rotate(textRotation); //textRotationis my variable
                                                watermark.Extent(resultSize.Width, resultSize.Height, gravity);

                                                image.Composite(watermark, gravity, CompositeOperator.Over);`

Thanks in advance,

**Comment by fmw42** [comment, marked as answer]
Add the equivalent of -colorspace sRGB just before the output.

**Reply by apandzo** [reply]
@fmw42 Hi Fred, thank you very much for such a prompt reply. It is working fine now. Much appreciated.
